Assets/Scripts/Scenes/InGame/Player/Item/ItemEffectScript/StatueOfRhanEffect.cs

```diff
@@ -0,0 +1,47 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UniRx;
+
+namespace Scenes.Ingame.Player
+{
+    public class StatueOfRhanEffect : ItemEffect
+    {
+        private bool _isHaveStatue = false;
+        public override void OnPickUp()
+        {
+            if (!ownerPlayerItem.IsHaveStatueEffect)
+            {
+                ownerPlayerItem.StartCoroutine("HealOfStatue");
+                ownerPlayerItem.ChangeHaveStatue(true);
+            }
+
+        }
+
+        public override void OnThrow()
+        {
+            //�@�A�C�e���X���b�g���ɐΑ����c���Ă��邩�ǂ�������
+            for (int i = 0 ; i < 7 ; i++) {
+                var item = ownerPlayerItem.ItemSlots[i];
+                if ( i != ownerPlayerItem.nowIndex && item.myItemData != null && item.myItemData.itemID == 23) 
+                {
+                     _isHaveStatue = true;
+                }
+            }
+            //����肪�c���Ă��Ȃ���Ό��ʏ���
+            if (!_isHaveStatue) 
+            {
+                ownerPlayerItem.StopCoroutine("HealOfStatue");
+                ownerPlayerItem.ChangeHaveStatue(false);
+            }          
+        }
+
+        public override void Effect()
+        {
+
+        }
+
```

Comment:
```suggestion
        private void OnDestroy()
        {
            _cancellationTokenSource.Cancel();
            _cancellationTokenSource.Dispose();
        }
```
ついでにアイテムが壊れるときにもキャンセルしてもらうようにする